pkg/transaction/transaction.go

```diff
@@ -385,13 +403,15 @@ func (t *transactionService) ResendTransaction(ctx context.Context, txHash commo
 		return err
 	}
 
-	tx := types.NewTx(&types.LegacyTx{
-		Nonce:    storedTransaction.Nonce,
-		To:       storedTransaction.To,
-		Value:    storedTransaction.Value,
-		Gas:      storedTransaction.GasLimit,
-		GasPrice: storedTransaction.GasPrice,
-		Data:     storedTransaction.Data,
+	tx := types.NewTx(&types.DynamicFeeTx{
+		Nonce:     storedTransaction.Nonce,
+		ChainID:   t.chainID,
+		To:        storedTransaction.To,
+		Value:     storedTransaction.Value,
+		Gas:       storedTransaction.GasLimit,
+		GasTipCap: storedTransaction.GasPrice,
+		GasFeeCap: storedTransaction.GasPrice,
```

Comment:
Shouldn't we be recalling the individual GasTipCap and GasFeeCap that were originally specified for the transaction?  Or haven't we modified the transaction store for the new fields yet?

Reply:
it's ok to put the gas price of the transaction as the gas fee cap and tip because the gas price was originally calculated using the gas tip cap and gas fee cap. 

here is an example:
https://goerli.etherscan.io/tx/0x621afbab180977bc77230ec1d5066bda0c617bd8a7126f4896c384ffb8b94dc7

where the gas price is base fee + gas tip, with an upper bound of gas fee cap.

ideally, what needs to be done is get a fresh suggested gas price and tip cap and resent the trx with that because the base price of the blockchain might have changed in the meanwhile.